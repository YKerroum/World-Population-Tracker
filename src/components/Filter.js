import React from 'react';
import PropTypes from 'prop-types';

const Filter = ({ handleSearch, tar }) => (
  <input placeholder={`Enter ${tar} to search`} className="searchInput" onChange={(e) => handleSearch(e.target.value)} />
);
Filter.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  tar: PropTypes.string.isRequired,
};

export default Filter;
