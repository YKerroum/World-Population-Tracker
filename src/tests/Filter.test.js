import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Filter from '../components/Filter';

test('Filter renders correctly', () => {
  // Assign
  const tree = renderer
    .create(
      <BrowserRouter>
        <Filter />
      </BrowserRouter>,
    )
    .toJSON();
    // Assert
  expect(tree).toMatchSnapshot();
});
