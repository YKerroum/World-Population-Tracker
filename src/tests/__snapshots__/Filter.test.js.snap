// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<input
  className="searchInput"
  onChange={[Function]}
  placeholder="Enter undefined to search"
/>
`;
